modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryEntriesExpireTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.query.continuous;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import javax.cache.expiry.CreatedExpiryPolicy;
+import javax.cache.expiry.Duration;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.cache.query.ContinuousQuery;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.GridCacheContext;
+import org.apache.ignite.internal.processors.continuous.GridContinuousProcessor;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.CU;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.testframework.GridTestUtils.getFieldValue;
+
+/**
+ * Test continuous queries with expired entries.
+ */
+public class CacheContinuousQueryEntriesExpireTest extends GridCommonAbstractTest {
+    /** @throws Exception If fails. */
+    @Test
+    public void testBackupQOnEntriesExpire() throws Exception {
+        IgniteEx srv0 = startGrid(0);
+        IgniteEx srv1 = startGrid(1);
+
+        IgniteCache<Object, Object> cache = srv0.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+                .setBackups(1)
+                .setAffinity(new RendezvousAffinityFunction().setPartitions(10)))
+            .withExpiryPolicy(new CreatedExpiryPolicy(new Duration(TimeUnit.MILLISECONDS, 1)));
+
+        srv1.cache(DEFAULT_CACHE_NAME).query(new ContinuousQuery<>().setLocalListener(evts -> {}).setIncludeExpired(true));
```

Comment:
this test is not clear for me, why do e need to set "setIncludeExpired" here ?
also issue is about expired items, but fix is about limitation for message sending, did i miss smth ?

Reply:
Because in the original ticket, the patch fixed the issue with the expired entity and has the tests. 
The issue appears after implementing some optimization for CQ (IGNITE-13126), but we didn't port it yet. In any case the test can help us in the future if we meet an issue like was introduced earlier in AI2.
Meanwhile GG-36974 have a few comments about these.

Reply:
Of course i have read this comments but as for me such approach is confusing, imagine that this test become flaky, someone will annotate code and found ... only these not relevant changes, btw did you check this test after IGNITE-13126 ? or we blindly port it ?

Reply:
yes, I've checked it